server/odc-service/src/main/java/com/oceanbase/odc/service/collaboration/project/ProjectService.java

```diff
@@ -182,14 +229,20 @@ public Project update(@NotNull Long id, @NotNull Project project) {
 
     @PreAuthenticate(hasAnyResourceRole = {"OWNER"}, resourceType = "ODC_PROJECT", indexOfIdParam = 0)
     @Transactional(rollbackFor = Exception.class)
-    public Project setArchived(Long id, @NotNull SetArchivedReq req) {
+    public Project setArchived(Long id, @NotNull SetArchivedReq req) throws InterruptedException {
         ProjectEntity previous = repository.findByIdAndOrganizationId(id, currentOrganizationId())
                 .orElseThrow(() -> new NotFoundException(ResourceType.ODC_PROJECT, "id", id));
         if (!req.getArchived()) {
             throw new BadRequestException("currently not allowed to recover projects");
         }
-        previous.setArchived(req.getArchived());
+        previous.setArchived(true);
         ProjectEntity saved = repository.save(previous);
+        List<ConnectionEntity> connectionEntities = connectionConfigRepository.findByProjectId(id);
+        for (ConnectionEntity entity : connectionEntities) {
+            entity.setProjectId(null);
+            connectionConfigRepository.save(entity);
+            connectionService.updateDatabaseProjectId(entity.getId(), null);
```

Comment:
What if the database sync schedule happens at the same time when you update database's related project id? Any concurrency issues?